detox/runners/jest/DetoxAdapterJasmine.js

```diff
@@ -1,9 +1,15 @@
 const _ = require('lodash');
 const DetoxAdapter = require('./DetoxAdapterImpl');
+const log = require('../../src/utils/logger').child({ __filename });
 
 class DetoxAdapterJasmine /* extends JasmineReporter */ {
   constructor(detox) {
     this._adapter = new DetoxAdapter(detox, DetoxAdapterJasmine._describeInitError);
+    this._printDeprecationWarning = _.once(() => {
+      log.warn(
+        '"jest-jasmine2" support is going to be dropped soon.\n' +
+        'Please follow the migration guide: https://github.com/wix/Detox/blob/master/docs/Guide.Migration.md#1630');
+    });
```

Comment:
I have to ask - is this really necessary? At what intermediate phase will we still have jasmine (pre-deprecation) Will it not be alright to kill it and release a major version?

Reply:
I think that the safest bet is to code-freeze old Jest integrations and gradually migrate to the new DetoxCircusEnvironment. After 1-2 months, we'll release a major version that drops the old Jest integrations.